pkg/ibmcsidriver/server.go

```diff
@@ -158,3 +160,39 @@ func logGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, h
 	}
 	return resp, err
 }
+
+func removeCSISocket(endPoint string) {
+	// Reference: https://github.com/kubernetes-csi/node-driver-registrar/blob/master/cmd/csi-node-driver-registrar/node_register.go#L168
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	<-sigc
+	err := os.Remove(endPoint)
+	if err != nil && !os.IsNotExist(err) {
+		glog.Errorf("failed to remove socket: %s with error: %+v", endPoint, err)
+		os.Exit(1)
+	}
+	/*
+		This is a temporary code to cleanup csi-socket created under csi-plugins directory.
+		This code must be removed once current supported versions are deprecated and
+		new major release is done.
+	*/
+	csiPluginSocketPath := "/lib-kubelet/csi-plugins/vpc.block.csi.ibm.io/"
+	if directoryExists(csiPluginSocketPath) {
+		err := os.RemoveAll(csiPluginSocketPath)
```

Comment:
can you check force removal option as well

Reply:
Checked. There is no such option.